vignettes/custom_data_sources.Rmd

```diff
@@ -0,0 +1,77 @@
+---
+title: "Using `geoknife` with a custom webgeom and/or webdata"
+author: "David Blodgett"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Vignette Title}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+```{r init, echo=FALSE, eval=TRUE, message=FALSE, warning=FALSE}
+library(geoknife)
+```
+
+This vignette shows how to use a custom Web Feature Service with `geoknife`. In this case, we are using a WFS from [ScienceBase](https://www.sciencebase.gov/catalog/item/54296bf0e4b0ad29004c2fbb). The url used for the WFS can be found in the "Spatial Services" section of the sciencebase item. If the WFS url you have has parameters included, remove them when passing them to `geoknife`.  
+e.g. this: `https://www.sciencebase.gov/catalogMaps/mapping/ows/54296bf0e4b0ad29004c2fbb`  
+not this: `https://www.sciencebase.gov/catalogMaps/mapping/ows/54296bf0e4b0ad29004c2fbb?service=wfs&request=getcapabilities&version=1.0.0`
+
+Since this is an advanced feature, it may be interesting to see what `geoknife` is doing, swith `verbose=FALSE` to `verbose=TRUE` in the following code to see the web service request being made when you execute this vignette.
+
+```{r stencil, echo=T, eval=T}
+gconfig(verbose=FALSE)
+stencil <- webgeom(url="https://www.sciencebase.gov/catalogMaps/mapping/ows/54296bf0e4b0ad29004c2fbb")
+
+stencil_geoms <- query(stencil, 'geoms')
+print(stencil_geoms)
+```
+
+Now we can select a WFS geometry layer (`geom`) and query for its attributes. 
+```{r attribute, echo=T, eval=T}
+geom(stencil) <- stencil_geoms[2]
+
+stencil_attributes <- query(stencil, 'attributes')
+print(stencil_attributes)
+```
+
+Now we can select an attribute and query for the values of that attribute. For this demo, we won't set the values, but rather just move forward using all polygons in this geom
+```{r value, echo=T, eval=T}
+attribute(stencil) <- stencil_attributes[2]
+
+print(query(stencil, 'values'))
+```
+
+Now we'll set up our webdata with an OPeNDAP service as it's url. Note that this can be any OPeNDAP service compatible with `geoknife` and the Geo Data Portal. What datasets are compatible is beyond the scope of this vignette, but is documented [here](https://my.usgs.gov/confluence/display/GeoDataPortal/Custom+Dataset+Use+Guidlines). The OPeNDAP dataset used here is from the [main USGS THREDDS archive](https://cida.usgs.gov/thredds) but many other OPeNDAP datasets can be used in the same way.
```

Comment:
Given you don't use a custom dataset for the `fabric`, do you want to rename this vignette to "custom spatial data sources"?